src/mock-assembly/MockAssembly.cs

```diff
@@ -315,4 +321,43 @@ public class GrandChildClass
             }
         }
     }
+
+    [TestFixture]
+    public class FixtureWithAttachment
+    {
+        public const int Tests = 2;
+
+        public static readonly string Attachment1Name = "attachment1.txt";
+        public static readonly string Attachment1Contents = "CONTENTS";
+        public static readonly string Attachment1Description = "A description with some <values>";
+
+        public static readonly string Attachment2Name = "attachment2.txt";
+        public static readonly string Attachment2Contents = "CONTENTS2";
+        public static readonly string Attachment2Description = "second description";
+
+        [Test]
+        public void SingleAttachmentTest()
+        {
+            var context = TestContext.CurrentContext;
+            var filepath = Path.Combine(context.WorkDirectory, Attachment1Name);
+
+            File.WriteAllText(filepath, Attachment1Contents);
+            TestContext.AddTestAttachment(filepath, Attachment1Description);
+        }
+
+        [Test]
+        public void MultiAttachmentTest()
+        {
+            var context = TestContext.CurrentContext;
+            var filepath1 = Path.Combine(context.WorkDirectory, Attachment1Name);
+            var filepath2 = Path.Combine(context.WorkDirectory, Attachment2Name);
+
+            File.WriteAllText(filepath1, Attachment1Contents);
+            TestContext.AddTestAttachment(filepath1, Attachment1Description);
+
+            File.WriteAllText(filepath2, Attachment2Contents);
+            TestContext.AddTestAttachment(filepath2, Attachment2Description);
```

Comment:
I am a bit hesitant to check in xml at the moment, since I don't know if the format is set in stone yet. nunit/docs#197 and nunit/nunit#1670 are still open but that is no problem?

The current approach was taken because I did not know nunit good enough to get the xml to persist.
I do believe it is valuable to have end-2-end tests for this functionality in case anything changes, but maybe it also makes sense with some more traditional unit tests such as without description etc?

1. For the files, it might make more sense to just use reference assemblies instead that whey no files need to be created or removed. What do you think?
2. For some extra unit tests it should be easy to add a new testfixture with new test but for the existing tests is there some recommended approach how to test the whole chain (running tests -> nunit xml -> TestResult) if they do not belong where they are?